lib/puppet/transaction/resource_harness.rb

```diff
@@ -99,10 +99,15 @@ def perform_changes(resource, context)
   def allow_changes?(resource)
     if resource.purging? and resource.deleting? and deps = relationship_graph.dependents(resource) \
             and ! deps.empty? and deps.detect { |d| ! d.deleting? }
-      deplabel = deps.collect { |r| r.ref }.join(",")
-      plurality = deps.length > 1 ? "":"s"
-      resource.warning "#{deplabel} still depend#{plurality} on me -- not purging"
-      false
+      real_deps = deps.reject { |r| r.class.name == :whit }
```

Comment:
I *think* the idea with doing deps as part of the if logic was to short-circuit walking the graph if we aren't purging. I agree this whole thing could use a cleanup, but I kinda feel it's outside the scope of this PR